server/src/main/java/org/elasticsearch/indices/IndicesQueryCache.java

```diff
@@ -121,12 +124,16 @@ private long getShareOfAdditionalRamBytesUsed(long cacheSize) {
         }
 
         final long additionalRamBytesUsed;
-        if (totalSize == 0) {
+        if (totalItemsInCache == 0) {
             // all shards have zero cache footprint, so we apportion the size of the shared bytes equally across all shards
             additionalRamBytesUsed = Math.round((double) sharedRamBytesUsed / shardCount);
         } else {
-            // some shards have nonzero cache footprint, so we apportion the size of the shared bytes proportionally to cache footprint
-            additionalRamBytesUsed = Math.round((double) sharedRamBytesUsed * cacheSize / totalSize);
+            /*
+             * some shards have nonzero cache footprint, so we apportion the size of the shared bytes proportionally to the number of
+             * segment-requests in the cache for this shard (the number and size of documents associated with those requests is irrelevant
+             * for this calculation).
```

Comment:
I'd like to put a hat on the notion that this is just a choice we've made. It's not like there's some from-on-high algorithm that we're implementing. This is just how we've chosen to do it.